integration/hsm/hsm_test.go

```diff
@@ -84,7 +83,7 @@ func etcdBackendConfig(t *testing.T) *backend.Config {
 	t.Cleanup(func() {
 		bk, err := etcdbk.New(context.Background(), cfg.Params)
 		require.NoError(t, err)
-		require.NoError(t, bk.DeleteRange(context.Background(), []byte(prefix),
+		require.NoError(t, bk.DeleteRange(context.Background(), append([]byte("/"), []byte(prefix)...),
```

Comment:
I don't believe this helps either, the internal prefix used by etcdbk is not involved in the actual backend operations, so `DeleteRange` cannot actually make use of it.

You want something like `DeleteRange(ctx, backend.Key() /* i.e. []byte("/") */, backend.RangeEnd(backend.Key()))` here, and even then you might be missing out on deleting things like `.locks/`. If we want to be very thorough, and if we believe the `allowPattern` used by the sanitizer wrapper, the earliest possible key is `+` and the latest possible key starts with `z`, so "delete everything" should be `DeleteRange(ctx, []byte("+"), backend.RangeEnd([]byte("z")))`.

Reply:
Don't know what I was thinking here, it just happened to work the first couple of times I tried it because it was completely wrong.... Should be fixed in the latest revision